### PR TITLE
[MINI-5628] Revamp analytics

### PR DESCRIPTION
**Pull request by laubengaier**
# Description
Adding trackings for example app
- `Environment Mode` and `Preview Mode` will not be tracked but later updated with moving them into the list configuration
- Version should be 1 since it's the build version not the short bundle version otherwise (before) it was displayed as `5.0.0-DEV (5.0.0)`
- Added protocol `ViewTrackable` and adoption to bring a convenient (swiftui) way to track analytics

## Links
https://jira.rakuten-it.com/jira/browse/MINI-5628

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

**Review by GondiTilak — APPROVED** [review]
Looks good 👍

File: Example/Controllers/SwiftUI/Features/Settings/Features/MiniAppSettingsAccessTokenView.swift [hunk 1]
```diff
@@ -69,6 +73,7 @@ struct MiniAppSettingsAccessTokenView: View {
             accessTokenString = tokenInfo.tokenString
             expiryDate = tokenInfo.expiryDate
         }
+        .trackPage(pageName: "Access Token")
```
Comment (rleojoseph) [inline Example/Controllers/SwiftUI/Features/Settings/Features/MiniAppSettingsAccessTokenView.swift:76]: We have few localization created for all screens, like `demo.app.rat.page.name.access_token` ? Please use the same

File: Example/Controllers/SwiftUI/Helper/View+Tracking.swift [hunk 2]
```diff
@@ -0,0 +1,127 @@
+import SwiftUI
+import MiniApp
+
+struct PageTrackingModifier: ViewModifier {
```
Comment (rleojoseph) [inline Example/Controllers/SwiftUI/Helper/View+Tracking.swift:4]: Do we have `RATViewController` still in use ?

**Review by rleojoseph — CHANGES_REQUESTED** [review]
Overall looks good to me. Just few changes required for localization.

File: Example/Controllers/SwiftUI/Features/Settings/Features/MiniAppSettingsAccessTokenView.swift (hunk 1 above)
Comment (laubengaier) [reply Example/Controllers/SwiftUI/Features/Settings/Features/MiniAppSettingsAccessTokenView.swift]: localization added

File: Example/Controllers/SwiftUI/Helper/View+Tracking.swift (hunk 2 above)
Comment (laubengaier) [reply Example/Controllers/SwiftUI/Helper/View+Tracking.swift:8]: not anymore I've removed the unnecessary analytics files

File: Example/Controllers/SwiftUI/Features/Settings/Features/MiniAppSettingsContactsView.swift [hunk 3]
```diff
@@ -46,9 +48,16 @@ struct MiniAppSettingsContactsView: View {
         .onAppear {
             contacts = viewModel.getContacts()
         }
+        .trackPage(pageName: pageName)
     }
 }
 
+extension MiniAppSettingsContactsView: ViewTrackable {
+	var pageName: String {
+		return NSLocalizedString("demo.app.rat.page.name.contactslist", comment: "")
```
Comment (rleojoseph) [inline Example/Controllers/SwiftUI/Features/Settings/Features/MiniAppSettingsContactsView.swift:57]: padding issue

File: Example/Controllers/SwiftUI/Features/Settings/Features/MiniAppSettingsContactsView.swift [hunk 4]
```diff
@@ -81,11 +90,12 @@ extension MiniAppSettingsContactsView {
         }
     }
 
-    struct ContactFormView: View {
+    struct ContactFormView: View, ViewTrackable {
 
         @Binding var name: String
         @Binding var contactId: String
         @Binding var email: String
+		@Binding var isPresented: Bool
```
Comment (rleojoseph) [inline Example/Controllers/SwiftUI/Features/Settings/Features/MiniAppSettingsContactsView.swift:98]: same padding

**Review by rleojoseph — APPROVED** [review]
Looks good to me :+1: thanks for updating the localization values.

**Pull request merged**